### PR TITLE
Fixed the bug that causes IntegerDivisionByZero

### PR DESCRIPTION
**Pull request by in74mz**

### CODE REVIEW

**Comment by serenader2014** [comment]
I see there are lots of changes in this commit, but many of them are just code formatting, and this makes me hard to review. Can you tell me which lines of codes do you change?

File: lib/carousel_slider.dart [hunk 1]
```diff
@@ -146,25 +148,35 @@ class CarouselSlider extends StatefulWidget {
   /// Jumps the page position from its current value to the given value,
   /// without animation, and without checking if the new value is in range.
   void jumpToPage(int page) {
-    final index = _getRealIndex(pageController.page.toInt(), realPage, items.length);
-    return pageController.jumpToPage(pageController.page.toInt() + page - index);
+    final index =
+        _getRealIndex(pageController.page.toInt(), realPage, items.length);
+    if (index == null) return;
```
Comment (in74mz) [inline lib/carousel_slider.dart]: here

File: lib/carousel_slider.dart [hunk 2]
```diff
@@ -146,25 +148,35 @@ class CarouselSlider extends StatefulWidget {
...
   ///
   /// The animation lasts for the given duration and follows the given curve.
   /// The returned [Future] resolves when the animation completes.
   Future<void> animateToPage(int page, {Duration duration, Curve curve}) {
-    final index = _getRealIndex(pageController.page.toInt(), realPage, items.length);
-    return pageController.animateToPage(pageController.page.toInt() + page - index,
-        duration: duration, curve: curve);
+    final index =
+        _getRealIndex(pageController.page.toInt(), realPage, items.length);
+    if (index == null) return Future<void>.value();
```
Comment (in74mz) [inline lib/carousel_slider.dart]: [L166] Added

File: lib/carousel_slider.dart [hunk 3]
```diff
@@ -221,18 +235,19 @@ class _CarouselSliderState extends State<CarouselSlider> with TickerProviderStat
...
       itemBuilder: (BuildContext context, int i) {
-        final int index =
-            _getRealIndex(i + widget.initialPage, widget.realPage, widget.items.length);
+        final int index = _getRealIndex(
+            i + widget.initialPage, widget.realPage, widget.items.length);
 
         return AnimatedBuilder(
           animation: widget.pageController,
-          child: widget.items[index],
+          child: index != null ? widget.items[index] : null,
```
Comment (in74mz) [inline lib/carousel_slider.dart]: [L250] Modified

File: lib/carousel_slider.dart [hunk 4]
```diff
@@ -284,6 +304,11 @@ int _getRealIndex(int position, int base, int length) {
...
 int _remainder(int input, int source) {
-  final int result = input % source;
-  return result < 0 ? source + result : result;
+  try {
+    final int result = input % source;
+    return result < 0 ? source + result : result;
+  } on IntegerDivisionByZeroException catch (e) {
+    print('[CarouselSlider] $e');
+    return null;
+  }
```
Comment (in74mz) [inline lib/carousel_slider.dart]: [L307-L313] Modified

**Comment by in74mz** [comment]
Ah, I'm sorry to confuse you.
These are the code lines I've modified.

[#L153](https://github.com/serenader2014/flutter_carousel_slider/pull/83/files#r357326568)
[#L166](https://github.com/serenader2014/flutter_carousel_slider/pull/83/files#r357327775)
[#L250](https://github.com/serenader2014/flutter_carousel_slider/pull/83/files#r357327983)
[#L307-L313](https://github.com/serenader2014/flutter_carousel_slider/pull/83/files#r357328238)

**Comment by serenader2014** [comment]
Can you open another PR which contains only these changes?

**Comment by AlexanderShniperson** [comment]
already done by pr #74

**Comment by AlexanderShniperson** [comment]
there is just check and return zero, but we should disscuss about `should we throw exception or just keep silent about this error`

**Comment by serenader2014** [comment]
I'm closing this issue since it was done in #74

**Pull request closed without merging**